### ISSUE TITLE
Loss becomes NaN after 20k steps

### ISSUE THREAD

**Issue by vss96** [experiencor/keras-yolo3#198]
So I was trying to train a separate set of images on this model and i noticed that after a certain amount of time the loss becomes NaN. I'm not sure why it is happening and i tried increasing the number of warmup epochs to 10 but that still didn't help.
My config file:
`{

    "model" : {
        "min_input_size":       100,
        "max_input_size":       600,
        "anchors":              [27,26, 59,55, 94,115, 147,68, 164,281, 182,163, 261,303, 335,183, 380,356],

        "labels":               ["redbull"]

   },

    "train": {
        "train_image_folder":   "../../data/synthetic_data_approach_1/images/",
        "train_annot_folder":   "../../data/synthetic_data_approach_1/annotations/",
        "cache_name":           "flickr_synth_train.pkl",
        "pretrained_weights":   "yolov3.weights",

        "train_times":          8,
        "batch_size":           6,
        "learning_rate":        1e-3,
        "nb_epochs":            50,
        "warmup_epochs":        10,
        "ignore_thresh":        0.5,
        "gpus":                 "0",

        "grid_scales":          [1,1,1],
        "obj_scale":            5,
        "noobj_scale":          1,
        "xywh_scale":           1,
        "class_scale":          1,

        "tensorboard_dir":      "logs",
        "saved_weights_name":   "logo_weights.h5",
        "debug":                true,
        "trainable":            true

    },

    "valid": {
        "valid_image_folder":   "",
        "valid_annot_folder":   "",
        "cache_name":           "",
        "valid_times":          1
    }
}`

**Comment by KtK99** [comment]
I am training the model on custom dataset, and the loss turned to nan after 4 iteration. The total loss before it turned to nan was 2.14... I am not sure what needs to be done at this point. 
What could be the source of this nan value?

**Comment by vss96** [comment]
@KtK99 Hi, I reduced the learning rate to 1e-4 and it worked out. Also, something else that you can try out is increasing the number of warmup epochs. Hope that helps :)

**Comment by KtK99** [comment]
@vss96 Thanks for your reply. I did think about the learning rate, if it's high enough it can diverge, thus we can get NAN, but I was following standard learning_rate value 1e-4. I suppose I could try with higher warm-up epochs. And I suppose it will be hard to get answers for custom data set as mine.

**Comment by vss96** [comment]
I would suggest increasing the size of the dataset. My guess is that the nan values show up because of gradient problems. You can always augment your dataset in different ways.
